base/heartbeat.go

```diff
@@ -198,66 +223,28 @@ func (h *couchbaseHeartBeater) checkStaleHeartbeats(staleThresholdMs int, handle
 
 			// doc not found, which means the heartbeat doc expired.
 			// call back the handler.
-			handler.StaleHeartBeatDetected(heartbeatDoc.NodeUUID)
+			handler.StaleHeartBeatDetected(heartbeatDocID)
 
 			// delete the heartbeat doc itself so we don't have unwanted
 			// repeated callbacks to the stale heartbeat handler
-			docId := h.heartbeatDocId(heartbeatDoc.NodeUUID)
-			if err := h.bucket.Delete(docId); err != nil {
-				Infof(KeyImport, "Failed to delete heartbeat doc: %v err: %v", docId, err)
-			}
-
+			h.heartbeatHandler.RemoveNode(heartbeatDocID)
 		}
 
 	}
 	h.checkCount++
 	return nil
 }
 
-func (h *couchbaseHeartBeater) heartbeatTimeoutDocId(nodeUuid string) string {
-	return fmt.Sprintf("%vheartbeat_timeout:%v", h.keyPrefix, nodeUuid)
-}
-
-func (h *couchbaseHeartBeater) heartbeatDocId(nodeUuid string) string {
-	return fmt.Sprintf("%vheartbeat:%v", h.keyPrefix, nodeUuid)
+func heartbeatTimeoutDocId(nodeUuid, keyPrefix string) string {
+	return fmt.Sprintf("%vheartbeat_timeout:%v", keyPrefix, nodeUuid)
```

Comment:
Switch to cheaper string concat

```
BenchmarkStrings/fmt.Sprintf-12   8298470    134 ns/op  48 B/op  3 allocs/op
BenchmarkStrings/str+str-12      57657954   20.3 ns/op   0 B/op  0 allocs/op
```

```suggestion
	return keyPrefix + "heartbeat_timeout:" + nodeUuid
```

Reply:
Done